rskj-core/src/main/java/co/rsk/mine/MinerClientInstantImpl.java

```diff
@@ -0,0 +1,114 @@
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package co.rsk.mine;
+
+import org.ethereum.rpc.TypeConverter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nonnull;
+import java.math.BigInteger;
+
+/**
+ * MinerClient for automine setting
+ *
+ * @author Ignacio Pulice
+ */
+public class MinerClientInstantImpl implements MinerClient {
+    private long nextNonceToUse = 0;
+    private static final Logger logger = LoggerFactory.getLogger("minerClient");
+    private final MinerServer minerServer;
+    private volatile boolean stop = false;
+    private volatile boolean isMining = false;
+    private volatile MinerWork work;
+
+    public MinerClientInstantImpl(MinerServer minerServer) {
+        this.minerServer = minerServer;
+    }
+
+    public void mine() {
+        isMining = true;
+    }
+
+    public boolean isMining() {
+        return this.isMining;
+    }
+
+    @Override
+    public boolean mineBlock() {
+        //if miner server was stopped for some reason, we don't mine.
+        if (stop) {
+            return false;
+        }
+
+        work = minerServer.getWork();
+
+        co.rsk.bitcoinj.core.NetworkParameters bitcoinNetworkParameters = co.rsk.bitcoinj.params.RegTestParams.get();
+        co.rsk.bitcoinj.core.BtcTransaction bitcoinMergedMiningCoinbaseTransaction = MinerUtils.getBitcoinMergedMiningCoinbaseTransaction(bitcoinNetworkParameters, work);
+        co.rsk.bitcoinj.core.BtcBlock bitcoinMergedMiningBlock = MinerUtils.getBitcoinMergedMiningBlock(bitcoinNetworkParameters, bitcoinMergedMiningCoinbaseTransaction);
+
+        BigInteger target = new BigInteger(1, TypeConverter.stringHexToByteArray(work.getTarget()));
+        boolean foundNonce = findNonce(bitcoinMergedMiningBlock, target);
+
+        if (foundNonce) {
+            logger.info("Mined block: " + work.getBlockHashForMergedMining());
+            minerServer.submitBitcoinBlock(work.getBlockHashForMergedMining(), bitcoinMergedMiningBlock);
+        }
+
+        return foundNonce;
+    }
+
+    @Override
+    public boolean fallbackMineBlock() {
+        //TODO: what should we do here ?
+        return false;
+    }
+
+    /**
+     * findNonce will try to find a valid nonce for bitcoinMergedMiningBlock, that satisfies the given target difficulty.
+     *
+     * @param bitcoinMergedMiningBlock bitcoinBlock to find nonce for. This block's nonce will be modified.
+     * @param target                   target difficulty. Block's hash should be lower than this number.
+     * @return true if a nonce was found, false otherwise.
+     * @remarks This method will return if the stop or newBetBlockArrivedFromAnotherNode intance variables are set to true.
+     */
+
+    private boolean findNonce(@Nonnull final co.rsk.bitcoinj.core.BtcBlock bitcoinMergedMiningBlock,
```

Comment:
This code is almost the same that is use in the other client, maybe we can move it to another place .